src/core/common/logging.cpp

```diff
@@ -96,13 +98,13 @@ static void Log(otLogLevel  aLogLevel,
             break;
         }
 
-#if OPENTHREAD_CONFIG_LOG_PREPEND_REGION
         IgnoreError(logString.Append("[%s]", levelStr));
-#endif
     }
 #endif // OPENTHREAD_CONFIG_LOG_PREPEND_LEVEL
 
+#if OPENTHREAD_CONFIG_LOG_PREPEND_REGION
     IgnoreError(logString.Append("%s", aRegionPrefix));
+#endif
```

Comment:
I think we may need this in general case (without `#if OPENTHREAD_CONFIG_LOG_PREPEND_REGION` check).

When `PREPEND_REGION` is not enabled but `PREPREND_LEVEL` is enabled, the prefix string becomes just `_OT_REGION_SUFFIX ` which would be `":"`. This is similar to how the macors work.

Reply:
updated